### ISSUE TITLE
[Mirth Administrator] Java Runtime Environment (JRE) - javaw.exe still in memory after stopping Mirth Administrator

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#737]
After using Mirth Administrator, when you stop it, the javaw.exe is not killed properly ; it is still in memory.
So if you run again Mirth Administrator, it uses a new instance of javaw.exe ... using again a new part of our memory ...
After a day if you use 10 time Mirth Administrator, you've 10 javaw.exe still in memory.

have to kill them manually ...

Is there something to change in the method of exiting in Mirth Administrator or is it Java not doing is job ?

Thanks for replies

Imported Issue. Original Details:
Jira Issue Key: MIRTH-730
Reporter: antaris
Created: 2008-01-25T02:50:32.000-0800

**Comment by rbeckman-nextgen** [comment]
I've noticed this issues as well.  I found that it only happens if you shut down the Mirth Administrator using the "Exit" button on the login-screen.  If you shut it down using the window close X button, it correclty exits the javaw.exe.  So as a simple work around, always exit using the window close button instead of the Exit button on the login screen.

I suspect this is just a problem of the action performed code for the Exit button doing a window close instead of a System.exit().  The javavm will not exit once you start the UI event thread running by creating the first window.  It must do a full System.exit() of the vm to make it shut down correctly.

Curt Welch
cwelch@vecna.com


Imported Comment. Original Details:
Author: cwelch
Created: 2008-02-04T08:52:03.000-0800
